### ISSUE TITLE
pip install with Python 3 fails

### ISSUE THREAD

**Issue by cdeil** [tisimst/mcerp#3]
Could you please make pip install work with Python 3 for `mcerp`?

```
$ pip install mcerp --user
Collecting mcerp
  Downloading mcerp-0.11.tar.gz
    Traceback (most recent call last):
      File "<string>", line 20, in <module>
      File "/private/var/folders/sb/4qv5j4m90pz1rw7m70rj1b1r0000gn/T/pip-build-8drpi_pi/mcerp/setup.py", line 13, in <module>
        import mcerp
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

      File "<string>", line 20, in <module>

      File "/private/var/folders/sb/4qv5j4m90pz1rw7m70rj1b1r0000gn/T/pip-build-8drpi_pi/mcerp/setup.py", line 13, in <module>

        import mcerp

    ----------------------------------------
    Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/sb/4qv5j4m90pz1rw7m70rj1b1r0000gn/T/pip-build-8drpi_pi/mcerp
```

I guess passing `use_2to3=True` to `setup.py` would be a quick fix, but a better one would be a common 2 / 3 code base?

**Comment by cdeil** [comment]
@tisimst – I wasn't able to find a link to the `mcerp` docs quickly (only via PyPI ... they're [here](http://pythonhosted.org//mcerp/index.html)) ... could you please add one to the Github front page and / or at the top of the README?

**Comment by chjimenezo92** [comment]
how did you install it?
I'm trying to use 2to3 in OS X system and there's an error

**Comment by cdeil** [comment]
> how did you install it?
> I'm trying to use 2to3 in OS X system and there's an error

@thianj23 - What do you mean? Are you working on a fix for this where you run `2to3` from `setup.py`?
If so, please make a pull request. I could then try it out and see if I get the same error / try to help make it work.

**Comment by paul-freeman** [comment]
I am using ``mcerp`` as a dependency in another project, and I really needed a Python 3 version. So I forked this project and created it. If anyone needs it, you can get a Python 3 version of this project by running

    pip install mcerp3

[Github link](https://github.com/paul-freeman/mcerp)
[PyPI link](https://pypi.org/project/mcerp3/1.0.0/)

If anyone notices any Python 3 issues, feel free to let me know and I'll try to fix them.

**Comment by cdeil** [comment]
@paul-freeman - Thanks!

@tisimst - Probably you don't have time or interest to maintain `mcerp` and support Python 3? Maybe giving @paul-freeman or @mkouhia (see #5) access so that development / maintenance can continue in one place is an option?

**Comment by tisimst** [comment]
I apologize for being so quiet. It isn't for lack of interest that I haven't done much. Just haven't had the time. I'm happy to give access to whomever would like to keep improving the code base. I always thought this package was useful and hope others feel the same. It would be a shame for it to fall by the wayside (or need to be forked) simply because I don't have the time to update it right now.

**Comment by paul-freeman** [comment]
I don't really have bandwidth to maintain much more than a downstream Python 3 build at the moment. But I will try to pull any patches that make their way into this repo. And if this one ever goes full Python 3 (or 4 even)... then I'll just make mine go away.

**Comment by cdeil** [comment]
@tisimst - I have a bit of time now, so if you give me access on Github, I could merge in the Python 3 support, put up Sphinx docs on readthedocs. If you have time to make a release, great; otherwise I'd also need access on PyPI (this is me: https://pypi.org/user/Christoph.Deil/).

@tisimst - Are you OK to put the BSD-3 license?

I see there are no automated tests, plus I'm not familiar with the method / codebase. But I guess using the examples from the docs and putting them as automated tests (run via pytest, set up travis-ci) would be a good start to put, to make sure Python 2 and 3 work.

@paul-freeman - I see this is what you did : https://github.com/tisimst/mcerp/compare/master...paul-freeman:master
With the renames it's not possible to merge this in. So I guess #5 can be use, or @paul-freeman if you can send a PR with any updates, we could merge that in.

**Comment by paul-freeman** [comment]
@cdeil - Happy to provide a PR with just the Python 3 code modifications. There is another small change for SciPy 1.0 that should be put in as well. If you get access to this repo from @tisimst let me know and I'll make the PR.

In the meantime, feel free to use my fork, mcerp3: [Conda](https://anaconda.org/freemapa/mcerp3) or [PyPI](https://pypi.org/project/mcerp3/)

**Comment by cdeil** [comment]
@paul-freeman - I now have access to the repo and some time to work on it.

If you could send a PR with the Python 3 fixes, and (preferably a separate) with the scipy 1.0 fix, that would be great!

**Comment by cdeil** [comment]
I just merged #7 - maybe that was already the fail you had with scipy 1.0.

**Comment by paul-freeman** [comment]
> I just merged #7 - maybe that was already the fail you had with scipy 1.0.

Yeah. This should fix it. I'll take a look at my fork and see how much it's diverged. I should be able to submit a pull request for the Python 3 stuff soon.

**Comment by paul-freeman** [comment]
Submitted the PR #10 for this just now.
